server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java

```diff
@@ -859,46 +856,68 @@ public void testScalingWithEmptyCoreAndKeepAliveAndWorkerPoolProbing() {
     }
 
     private void testScalingWithEmptyCore(EsThreadPoolExecutor executor) {
-        try {
-            class Task extends AbstractRunnable {
-                private int remaining;
-                private final CountDownLatch doneLatch;
-
-                Task(int iterations, CountDownLatch doneLatch) {
-                    this.remaining = iterations;
-                    this.doneLatch = doneLatch;
-                }
 
-                @Override
-                public void onFailure(Exception e) {
-                    fail(e);
-                }
+        class TaskScheduler extends SubscribableListener<Void> {
```

Comment:
Does it really need `extends SubscribableListener`? Composition seems preferable to inheritance here.